src/Microsoft.TestPlatform.Utilities/CodeCoverageDataAttachmentsHandler.cs

```diff
@@ -106,15 +112,25 @@ private async Task<string> MergeCodeCoverageFilesAsync(IList<string> files, IPro
 
         private async Task<string> MergeCodeCoverageFilesAsync(IList<string> files, CancellationToken cancellationToken)
         {
-            var coverageUtility = new CoverageFileUtility();
-
-            var coverageData = await coverageUtility.MergeCoverageFilesAsync(
-                    files,
-                    cancellationToken).ConfigureAwait(false);
-
-            coverageUtility.WriteCoverageFile(files[0], coverageData);
-
-            foreach(var file in files.Skip(1))
+            cancellationToken.ThrowIfCancellationRequested();
+
```

Comment:
Maybe add comment why we are invoking it dynamically. 
Also add try catch block inside this method or when you call MergeCodeCoverageFilesAsync - return original attachments in case of error